ferrypick.py

```diff
@@ -23,53 +27,87 @@ def parse_link(link):
     raise ValueError("Unrecognized link")
 
 
-def rename(bytesline, original_name, current_name):
+def rename(content, original_name, current_name):
     """
     On a given bytes line, naïvely replace original package name with current package name.
-    Works on pkgname.spec and pkgname.rpmlintrc only (as defined in REPLACE_SUFFIXES).
+    Works on pkgname.spec and pkgname.rpmlintrc only (as defined in SUFFIXES).
     """
-    if original_name != current_name:
-        for suffix in REPLACE_SUFFIXES:
-            for prefix in "a", "b":  # this is what git does
-                original = f"{prefix}/{original_name}.{suffix}".encode("utf-8")
-                current = f"{prefix}/{current_name}.{suffix}".encode("utf-8")
-                bytesline = bytesline.replace(original, current)
-    return bytesline
-
-
-@contextlib.contextmanager
-def patch(link, original_name, current_name):
+    if original_name is not None and original_name == current_name:
+        return content
+
+    def replace(regs):
+        prefix = regs.group('prefix')
+        suffix = regs.group('suffix')
+        return f"{prefix}/{current_name}.{suffix}"
+
+    if original_name is not None:
+        name = re.escape(original_name)
+    else:
+        name = FILENAME_RE
+    original = (r"(?P<prefix>%s)/%s\.(?P<suffix>%s)"
+                % ('|'.join(PREFIXES),
+                   name,
+                   '|'.join(SUFFIXES)))
```

Comment:
It gives something like `a/python.spec` but in a more generic form to match also "b/" prefix or ".rpmlint" suffix :-)

At the end, it's closer to `(a|b)/python\.(spec|rpmlint)`.

Regex are fun!

Reply:
It's actually the string modulo here that makes it hard to read for me. What about storing the prefixes regex-snippet and the suffixes regex-snippet in a constant and using an f-string here:

```python
rf"(?P<prefix>{PREFIXES_RE})/{name}\.(?P<suffix>{SUFFIXES_RE})"
```

Also, while converting this to regex this, could you wrap it in `\b`s? I haven't done that before only because I avoided regex.